File: llm_generated_code/ChatGPT/CPP/Level1/Q35.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, k, f;
        cin >> n >> k >> f;
        vector<int> cubes(n);
        for (int i = 0; i < n; ++i) {
            cin >> cubes[i];
        }

        vector<int> sorted_cubes = cubes;
        sort(sorted_cubes.rbegin(), sorted_cubes.rend());

        int favorite_value = cubes[f - 1];
        int top_k_value = sorted_cubes[k - 1];

        if (favorite_value > top_k_value) {
            cout << "NO" << endl;
        } else if (favorite_value < top_k_value) {
            cout << "YES" << endl;
        } else {
            cout << "MAYBE" << endl;
        }
    }
    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level1/Q35.py
def process_test_cases(t, test_cases):
    for case in test_cases:
        n, k, f, cubes = case
        sorted_cubes = sorted(cubes, reverse=True)
        favorite_cube = cubes[f]
        top_k_cubes = sorted_cubes[:k]
        
        if favorite_cube in top_k_cubes:
            print("YES")
        elif favorite_cube not in sorted_cubes[:k]:
            print("NO")
        else:
            print("MAYBE")

t = int(input())
test_cases = []
for _ in range(t):
    n, k, f = map(int, input().split())
    cubes = list(map(int, input().split()))
    test_cases.append((n, k, f, cubes))

process_test_cases(t, test_cases)
